app/src/main/java/com/github/sdpteam15/polyevents/ItemsAdminActivity.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.sdpteam15.polyevents
+
+import android.content.Context
+import android.os.Bundle
+import android.transition.Slide
+import android.transition.TransitionManager
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.widget.EditText
+import android.widget.ImageButton
+import android.widget.LinearLayout
+import android.widget.PopupWindow
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.github.sdpteam15.polyevents.database.Database.currentDatabase
+import com.github.sdpteam15.polyevents.item_request.ItemAdapter
+
+/**
+ * Activity displaying Items and supports items creation and deletion
+ */
+class ItemsAdminActivity : AppCompatActivity() {
+
+    //var items: MutableLiveData<MutableList<String>> = MutableLiveData()
+
+    /**
+     * Recycler containing all the items
+     */
+    lateinit var recyclerView: RecyclerView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_items_admin)
+
+        // TODO take existing items from the database
+
+        //val clickListener = { _: String -> } // TODO define what happens when we click on an Item
+        recyclerView = findViewById(R.id.id_recycler_items_list)
+        recyclerView.adapter = ItemAdapter(/*clickListener*/)
+
+        val btnAdd = findViewById<ImageButton>(R.id.id_add_item_button)
+        btnAdd.setOnClickListener { createAddItemPopup() }
+    }
+
+    private fun createAddItemPopup() {
```

Comment:
Minor (the function is well commented so this is clearly understandable): If the content of this function can be split into many smaller functions, this could increase modularity and readability.